pandas/_libs/algos_common_helper.pxi.in

```diff
@@ -38,39 +38,41 @@ def diff_2d(ndarray[diff_t, ndim=2] arr,
         #  warnings about unreachable code
         sx, sy = (<object>arr).shape
         if arr.flags.f_contiguous:
-            if axis == 0:
-                if periods >= 0:
-                    start, stop = periods, sx
+            with nogil:
```

Comment:
Can the `nogil` here not just precede the `if...else`?

Reply:
That'd be nice, but no.  The `arr.flags.f_contiguous` check is in python-space

Reply:
Can you assign to a bool before the block and check that at the top instead of staying in the Python space? On first impression I was expecting the difference branches to have different GIL considerations, hence the ask

Reply:
good idea, will do